### PR TITLE
Added ability to rename across file systems

### PR DESCRIPTION
**Pull request by gapra-msft**

### CODE REVIEW

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeDirectoryAsyncClient.java [hunk 1]
```diff
@@ -424,10 +428,11 @@ public DataLakeDirectoryAsyncClient getSubdirectoryAsyncClient(String subdirecto
      * @return A {@link Mono} containing a {@link Response} whose {@link Response#getValue() value} contains a {@link
      * DataLakeDirectoryAsyncClient} used to interact with the directory created.
      */
-    public Mono<Response<DataLakeDirectoryAsyncClient>> renameWithResponse(String destinationPath,
-        DataLakeRequestConditions sourceRequestConditions, DataLakeRequestConditions destinationRequestConditions) {
+    public Mono<Response<DataLakeDirectoryAsyncClient>> renameWithResponse(String fileSystemName,
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakeDirectoryAsyncClient.java]: change to destinationFileSystem

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # Release History
 
+## 12.0.1-beta.2 (2020-01-15)
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: Should this be changed to `12.0.1-beta.1`?

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java [hunk 3]
```diff
@@ -676,17 +676,22 @@ public DataLakeServiceVersion getServiceVersion() {
     /**
      * Package-private rename method for use by {@link DataLakeFileAsyncClient} and {@link DataLakeDirectoryAsyncClient}
      *
+     * @param destinationFileSystem The file system of the destination within the account.
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java]: Any reason this comment doesn't state that `null` will use the current file system as the other comments have?

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java [hunk 4]
```diff
@@ -719,32 +724,38 @@ public DataLakeServiceVersion getServiceVersion() {
...
-     * Takes in a destination path and creates a DataLakePathAsyncClient with a new path name
+     * Takes in a destination and creates a DataLakePathAsyncClient with a new path
+     * @param destinationFileSystem The destination file system
      * @param destinationPath The destination path
      * @return A DataLakePathAsyncClient
      */
-    DataLakePathAsyncClient getPathAsyncClient(String destinationPath) {
+    DataLakePathAsyncClient getPathAsyncClient(String destinationFileSystem, String destinationPath) {
+        if (CoreUtils.isNullOrEmpty(destinationFileSystem)) {
+            throw logger.logExceptionAsError(new IllegalArgumentException(
```
Comment (alzimmermsft) [inline sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java]: Any reason this doesn't default `destinationFileSystem` to the current file system? Looks like `renameWithResponse` will pass the current file system as the `destinationFileSystem` value. Seems odd to allow null in one case and not in another very similar case.

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: Yeah, I'll change it once we decide if we need to release again or not.

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java (hunk 3 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java]: It's internal so I didn't bother changing it, I can change it though.

File: sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java (hunk 4 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/src/main/java/com/azure/storage/file/datalake/DataLakePathAsyncClient.java]: Sure, I can change that

File: sdk/storage/azure-storage-file-datalake/CHANGELOG.md (hunk 2 above)
Comment (gapra-msft) [reply sdk/storage/azure-storage-file-datalake/CHANGELOG.md]: resolved

**Pull request merged**